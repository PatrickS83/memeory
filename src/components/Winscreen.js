import React from 'react';
import PropTypes from 'prop-types';

const Winscreen = props => (
  <React.Fragment>
    <h1 className="title has-text-centered">YOU WON !!!!</h1>
    <div className="columns is-multiline is-mobile">
      <div className="column is-6">
        <figure className="image is-square">
          <img src="https://media.giphy.com/media/gsrkIo59dnI8E/giphy.gif" alt="" />
        </figure>
      </div>
      <div className="column is-6">
        <figure className="image is-square">
          <img src="https://media.giphy.com/media/T6l1LBKmtUu9a/giphy.gif" alt="" />
        </figure>
      </div>
      <button className="button is-info playbutton" onClick={props.resetGame}>
        Play again!
      </button>
    </div>
  </React.Fragment>
);

Winscreen.propTypes = {
  resetGame: PropTypes.func.isRequired
};

export default Winscreen;
