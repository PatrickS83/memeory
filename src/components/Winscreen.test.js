import React from 'react';
import { shallow } from 'enzyme';
import Winscreen from './Winscreen';

describe('<Winscreen/>', () => {
  let wrapper;
  let resetGame;
  beforeEach(() => {
    resetGame = jest.fn();
    wrapper = shallow(<Winscreen resetGame={resetGame} />);
  });
  it('renders without crashing', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('renders the Winscreen as saved in snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('invokes resetGame() on reset-button click', () => {
    expect(resetGame).not.toHaveBeenCalled();
    wrapper.find('.playbutton').simulate('click');
    expect(resetGame).toHaveBeenCalled();
  });
});
