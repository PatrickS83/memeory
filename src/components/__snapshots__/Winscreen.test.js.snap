// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Winscreen/> renders the Winscreen as saved in snapshot 1`] = `
<React.Fragment>
  <h1
    className="title has-text-centered"
  >
    YOU WON !!!!
  </h1>
  <div
    className="columns is-multiline is-mobile"
  >
    <div
      className="column is-6"
    >
      <figure
        className="image is-square"
      >
        <img
          alt=""
          src="https://media.giphy.com/media/gsrkIo59dnI8E/giphy.gif"
        />
      </figure>
    </div>
    <div
      className="column is-6"
    >
      <figure
        className="image is-square"
      >
        <img
          alt=""
          src="https://media.giphy.com/media/T6l1LBKmtUu9a/giphy.gif"
        />
      </figure>
    </div>
    <button
      className="button is-info playbutton"
      onClick={[MockFunction]}
    >
      Play again!
    </button>
  </div>
</React.Fragment>
`;
